plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/UnsafeIntrinsics.java

```diff
@@ -102,25 +103,25 @@ private static void registerCompareAndExchangeIntrinsics(final CompilationContex
         // todo: research to discover if some platforms handle misalignment correctly, make it switchable
 
         intrinsics.registerIntrinsic(unsafeDesc, "compareAndExchangeInt", getCompareAndExchangeDesc(classContext, BaseTypeDescriptor.I),
-            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, VOLATILE, STRONG));
+            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, GlobalAcquire, STRONG));
```

Comment:
According to the equivalent operations on `VarHandle`, this method should have `GlobalSeqCst` on write and `SinglePlain` (or `GlobalPlain`) on read.